### PR TITLE
Libnotify support for linux client

### PR DESCRIPTION
**Pull request by ardadem**
I added basic libnotify support for linux client. It's optional like mysql (now it's not). Also we can move macos notifications to notfications.cpp in the future.

### CODE REVIEW

File: src/engine/client/notifications.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#ifndef ENGINE_CLIENT_NOTIFICATIONS_H
+#define ENGINE_CLIENT_NOTIFICATIONS_H
+void NotificationsInit();
```
Comment (Learath2) [inline src/engine/client/notifications.h]: I think all these functions belong in a class

File: src/engine/client/client.cpp [hunk 2]
```diff
@@ -3776,6 +3777,12 @@ void CClient::LoadFont()
 		m_pConsole->Print(IConsole::OUTPUT_LEVEL_STANDARD, "gameclient", "failed to load font. filename='%s'", pFontFile);
 }
 
+void CClient::Notify(const char *pTitle, const char *pMessage)
+{
+	NotificationsNotify(pTitle, pMessage);
+	Graphics()->NotifyWindow();
```
Comment (Learath2) [inline src/engine/client/client.cpp]: Notify window is a weird function, on macOS it's not implemented but the platform client does  `[NSApp requestUserAttention:NSInformationalRequest];` which is the same thing.
It'd be nice if you could clean this up while at it.

**Comment by Learath2** [comment]
Actually the macOS notifications won't work on newer macOS, you need authorization from the user to send them. I'll fix that later.

**Comment by heinrich5991** [comment]
https://github.com/ddnet/ddnet/pull/2099#discussion_r399678967 is still outstanding.

**Comment by heinrich5991** [comment]
If you need help with that or don't know what to do, just comment on this PR.

**Comment by Learath2** [comment]
@ardadem Can you move `notifications.mm` to `notifications.m`. It's no longer uses any ObjectiveC++ features so we might as well compile it as ObjectiveC.

Other than that LGTM

**Pull request merged**